src/syscalls/interfaces/mod.rs

```diff
@@ -117,55 +116,55 @@ pub trait SyscallInterface: Send + Sync {
 	}
 
 	fn get_mac_address(&self) -> Result<[u8; 6], ()> {
-		irqsave(|| match arch::kernel::pci::get_network_driver() {
-			Some(driver) => Ok(driver.borrow().get_mac_address()),
+		match arch::kernel::pci::get_network_driver() {
+			Some(driver) => Ok(driver.lock().get_mac_address()),
 			_ => Err(()),
-		})
+		}
 	}
 
 	fn get_mtu(&self) -> Result<u16, ()> {
-		irqsave(|| match arch::kernel::pci::get_network_driver() {
-			Some(driver) => Ok(driver.borrow().get_mtu()),
+		match arch::kernel::pci::get_network_driver() {
+			Some(driver) => Ok(driver.lock().get_mtu()),
 			_ => Err(()),
-		})
+		}
 	}
 
 	fn has_packet(&self) -> bool {
-		irqsave(|| match arch::kernel::pci::get_network_driver() {
-			Some(driver) => driver.borrow().has_packet(),
+		match arch::kernel::pci::get_network_driver() {
+			Some(driver) => driver.lock().has_packet(),
 			_ => false,
-		})
+		}
 	}
 
 	fn get_tx_buffer(&self, len: usize) -> Result<(*mut u8, usize), ()> {
-		irqsave(|| match arch::kernel::pci::get_network_driver() {
-			Some(driver) => driver.borrow_mut().get_tx_buffer(len),
+		match arch::kernel::pci::get_network_driver() {
+			Some(driver) => driver.lock().get_tx_buffer(len),
```

Comment:
On all of these driver.lock() calls you have the same issue as above that deadlocks might happen. A lot more likely to occur here, since these calls might happen frequently.

Reply:
see previous comment